server/services/conditionsService.js

```diff
@@ -336,3 +362,16 @@ function getSubmissionForCondition(conditionId, inputtedConditions) {
 
   return inputtedConditions[conditionId]
 }
+
+function constructConditionTexts(conditions) {
+  const formattedConditions = []
+  conditions.forEach((condition) => {
+    let contentText = ''
+    condition.forEach((conditionObject) => {
+      contentText += conditionObject.text || conditionObject.variable
+    })
+    formattedConditions.push({ text: contentText })
+  })
```

Comment:
code without mutation can be easier to read and reason about.
can you think how to refactor this to use map and join?

Reply:
amended